pkg/authorization/path/path.go

```diff
@@ -18,71 +18,46 @@ package path
-	for _, p := range inputPaths {
-		p = strings.TrimPrefix(p, "/")
-		if len(p) == 0 {
-			// matches "/"
-			paths.Insert(p)
-			continue
-		}
-		if strings.ContainsRune(p, '*') {
-			patterns = append(patterns, p)
-		} else {
-			paths.Insert(p)
-		}
-	}
-
-	return &pathAuthorizer{
-		matchDecision:   onMatch,
-		noMatchDecision: onNoMatch,
-		paths:           paths,
-		pathPatterns:    patterns,
-	}
+	delegatedPathAuthorizer authorizer.Authorizer
 }
 
 func (a *pathAuthorizer) Authorize(ctx context.Context, attr authorizer.Attributes) (authorizer.Decision, string, error) {
-	pth := strings.TrimPrefix(attr.GetPath(), "/")
-	if a.paths.Has(pth) {
-		return a.matchDecision, "", nil
-	}
+	decision, reason, err := a.delegatedPathAuthorizer.Authorize(ctx, attr)
 
-	for _, pattern := range a.pathPatterns {
-		if found, err := path.Match(pattern, pth); err != nil {
-			return authorizer.DecisionNoOpinion, "Error", err
-		} else if found {
-			return a.matchDecision, "", nil
-		}
+	// There is a match on the path, so we have no opinion and let subsequent
+	// authorizers in a union decide.
+	if err == nil && decision == authorizer.DecisionAllow {
+		return authorizer.DecisionNoOpinion, reason, nil
 	}
 
-	return a.noMatchDecision, "", nil
+	return authorizer.DecisionDeny, fmt.Sprintf("NOT(%s)", reason), err
 }
 
-func NewAllowPathAuthorizer(allowPaths []string) authorizer.Authorizer {
-	if len(allowPaths) == 0 {
-		return authorizer.AuthorizerFunc(func(context.Context, authorizer.Attributes) (authorizer.Decision, string, error) {
-			return authorizer.DecisionNoOpinion, "", nil
-		})
+// NewAllowedPathsAuthorizer returns an authorizer that has no opinion on
+// matches for the given paths and denies all others. No opinion ensures that
+// subsequent authorizers in a union are still called. The given paths can
+// include simple glob patterns.
+func NewAllowedPathsAuthorizer(allowPaths []string) (authorizer.Authorizer, error) {
+	delegatedPathAuthorizer, err := pathauthorizer.NewAuthorizer(allowPaths)
+	if err != nil {
+		return nil, fmt.Errorf("error creating path authorizer: %v", err)
 	}
-	return newPathAuthorizer(authorizer.DecisionNoOpinion, authorizer.DecisionDeny, allowPaths)
+
+	return &pathAuthorizer{delegatedPathAuthorizer: delegatedPathAuthorizer}, nil
 }
 
-func NewAlwaysAllowPathAuthorizer(alwaysAllowPaths []string) authorizer.Authorizer {
-	return newPathAuthorizer(authorizer.DecisionAllow, authorizer.DecisionNoOpinion, alwaysAllowPaths)
+// NewPassthroughAuthorizer returns an authorizer that allows on matches for
+// the given paths and has no opinion on all others. This allows skipping
+// subsequent authorizers in a union, effectively passing through the given
+// paths.
+// The given paths can include simple glob patterns.
+func NewPassthroughAuthorizer(ignorePaths []string) (authorizer.Authorizer, error) {
+	return pathauthorizer.NewAuthorizer(ignorePaths)
```

Comment:
this cuts down on functionality

Reply:
Yes, it does. It is based on your original concept. I wasn't sure if it is a good idea to drop that functionality, but @s-urbaniak said it is fine.